pkg/apis/org/v1/che_types.go

```diff
@@ -19,29 +19,34 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// +k8s:openapi-gen=true
 // CheClusterSpec defines the desired state of CheCluster
 type CheClusterSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	Server   CheClusterSpecServer  `json:"server"`
 	Database CheClusterSpecDB      `json:"database"`
 	Auth     CheClusterSpecAuth    `json:"auth"`
 	Storage  CheClusterSpecStorage `json:"storage"`
-	K8SOnly  CheClusterSpecK8SOnly `json:"k8s"`
+
+	// +optional
+	K8s CheClusterSpecK8SOnly `json:"k8s"`
 }
 
+// +k8s:openapi-gen=true
 type CheClusterSpecServer struct {
 	// AirGapContainerRegistryHostname is the hostname to the internal registry to pull images from in the air-gapped environment
 	AirGapContainerRegistryHostname string `json:"airGapContainerRegistryHostname"`
 	// AirGapContainerRegistryOrganization is the repository name in the registry to pull images from in the air-gapped environment
 	AirGapContainerRegistryOrganization string `json:"airGapContainerRegistryOrganization"`
 	// CheImage is a server image used in Che deployment
+	// +optional
```

Comment:
in addition to the `// +optional` comment, you can add the `,omitempty` marker in the json tag, like that:
```
	CheImage string `json:"cheImage,omitempty"`
```
Of course, only on values that will have defauts, which is the case for most of them.